homeassistant/components/otbr/websocket_api.py

```diff
@@ -59,32 +61,73 @@ async def websocket_info(
     assert border_agent_id is not None
     connection.send_result(
         msg["id"],
-        {
-            "active_dataset_tlvs": dataset_tlvs.hex() if dataset_tlvs else None,
-            "border_agent_id": border_agent_id.hex(),
-            "channel": dataset.channel if dataset else None,
-            "extended_address": extended_address.hex(),
-            "url": data.url,
-        },
+        [
+            {
+                "active_dataset_tlvs": dataset_tlvs.hex() if dataset_tlvs else None,
+                "border_agent_id": border_agent_id.hex(),
+                "channel": dataset.channel if dataset else None,
+                "extended_address": extended_address.hex(),
+                "url": data.url,
+            }
+        ],
     )
 
 
+def async_get_otbr_data(
+    orig_func: Callable[
+        [HomeAssistant, websocket_api.ActiveConnection, dict, OTBRData],
+        Coroutine[Any, Any, None],
+    ],
+) -> Callable[
+    [HomeAssistant, websocket_api.ActiveConnection, dict], Coroutine[Any, Any, None]
+]:
+    """Decorate function to get OTBR data."""
+
+    @wraps(orig_func)
+    async def async_check_border_agent_id_func(
+        hass: HomeAssistant, connection: websocket_api.ActiveConnection, msg: dict
+    ) -> None:
+        """Fetch OTBR data and pass to orig_func."""
+        if DOMAIN not in hass.data:
+            connection.send_error(msg["id"], "not_loaded", "No OTBR API loaded")
+            return
+
+        data: OTBRData = hass.data[DOMAIN]
+
+        try:
+            border_agent_id = await data.get_border_agent_id()
+        except HomeAssistantError as exc:
+            connection.send_error(msg["id"], "get_border_agent_id_failed", str(exc))
+            return
+
+        # The border agent ID is checked when the OTBR config entry is setup,
+        # we can assert it's not None
+        assert border_agent_id is not None
+        if border_agent_id.hex() != msg["border_agent_id"]:
+            connection.send_error(msg["id"], "unknown_router", "")
+            return
+
+        await orig_func(hass, connection, msg, data)
+
+    return async_check_border_agent_id_func
+
+
 @websocket_api.websocket_command(
     {
         "type": "otbr/create_network",
+        vol.Required("border_agent_id"): str,
```

Comment:
Initially, the idea was to use border agent id, which would be the nicer option.

However, as we've discovered recently Apple doesn't supply the `border_agent_id` currently. It doesn't matter here, but it meant we changed the Thread WS APIs to use `extended_address` instead.

Ontop of that,  we have known OTBR implementation which do not support `border_agent_id` (namely ESP32).

So with this two data points, I think it is better to just use `extended_address` here too.

Reply:
I'll change it, I think this PR was opened before we changed to use `extended_address` in `thread`